ckanext/dcatapchharvest/tests/test_swiss_schemaorg_profile_serialize.py

```diff
@@ -184,3 +190,41 @@ def test_graph_from_dataset(self):
             eq_(len([t for t in g.triples((dataset_ref, item[1], None))]), len(values))
             for value in values:
                 assert self._triple(g, dataset_ref, item[1], item[2](value))
+
+    def test_graph_from_dataset_uri(self):
+        """"Tests that datasets with a uri from the test system
+        have that uri changed to reference the prod system when they are output as a graph"""
+
+        dataset = {
+            'id': '4b6fe9ca-dc77-4cec-92a4-55c6624a5bd6',
+            'name': 'test-dataset',
+            'title': 'Test DCAT dataset',
+            'uri': 'https://test.example.com/dataset/foo',
+            'version': '1.0b',
+            'metadata_created': '2015-06-26T15:21:09.034694',
+            'metadata_modified': '2015-06-26T15:21:09.075774',
+            'resources': [
+                {'uri': 'https://test.example.com/dataset/foo', 'id': '5f2be71f-636c-4d3f-aac1-50830b97f853'},
+            ]
+        }
+
+        s = RDFSerializer(profiles=['swiss_schemaorg'])
+        g = s.g
+        dataset_ref = s.graph_from_dataset(dataset)
+
+        # Change dataset uri that includes a test url
+        dataset_uri = dh.dataset_uri(dataset, dataset_ref)
+        dataset_ref_changed = URIRef(dataset_uri)
+
+        # To test that the distribution is present in the graph with the new resource uri
+        for resource_dict in dataset.get("resources", []):
+            distribution = URIRef(dh.resource_uri(resource_dict))
+            # adding distribution (triples) to the graph
+            g.add((dataset_ref_changed, SCHEMA.distribution, distribution))
+            g.add((distribution, RDF.type, SCHEMA.Distribution))
```

Comment:
I'm sorry I missed this earlier - I must have scrolled down to the bottom and just looked at the `assert` for the distribution.

As we talked about earlier, you don't have to add the distribution to the graph. That already happens when the graph is serialised. If you add the distribution to the graph and then check that the distribution is in the graph, you're guaranteed to get a positive result no matter what the rest of the code is doing. ;)

You can remove this whole section and the test should still work.

Reply:
Using a different uri for the resource, compared to the dataset uri. 
Get the distribution with the dh.resource_uri helper function, which change the resource uri that contains a test url. 
Add another test `assert self._triple(g, distribution, RDF.type, SCHEMA.Distribution)`